bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -1154,6 +1159,73 @@ public static boolean format(ServerConfiguration conf,
         }
     }
 
+    /**
+     * Initializes bookie, by making sure that the journalDir, ledgerDirs and
+     * indexDirs are empty and there is no registered Bookie with this BookieId.
+     *
+     * @param conf
+     * @return
+     * @throws Exception
+     */
+    public static boolean initBookie(ServerConfiguration conf) throws Exception {
+        try (RegistrationManager rm = RegistrationManager.instantiateRegistrationManager(conf)) {
+            /*
+             * make sure that journalDir, ledgerDirs and indexDirs are empty
+             */
+            File[] journalDirs = conf.getJournalDirs();
+            for (File journalDir : journalDirs) {
```

Comment:
line 1176-1182, 1185-1192, 1196-1204 are having duplicated the logic. is it better to abstract this logic into a util function, rather than duplicating the code?